lib/headlines/security_headers/content_security_policy.rb

```diff
@@ -1,16 +1,109 @@
 module Headlines
   module SecurityHeaders
     class ContentSecurityPolicy < SecurityHeader
-      def parse
-        results = {}
-        value.split(";").each do |parameter|
-          key = parameter.split(" ")[0].gsub("-", "_")
-          value = parameter.split(" ")[1..-1].join(" ")
-          results[key.to_sym] = value
+      def initialize(headers, body, url)
+        @name = "content-security-policy"
+        @value = headers["content-security-policy"] || ""
+        @headers = headers
+        @body = body
+        @url = url
+      end
+
+      def score
+        valid? ? tests.sum { |t| t[:score] } : -15
```

Comment:
Can we avoid use of magic and reference something like CSP_RULES[:no_csp_header] ?